Test of the GUI failed on OSX python2.7

Jeromes-MacBook-Pro:silx kieffer$ python setup.py build test build_doc
missing cimport in module 'specfile': silx/io/specfile/specfile.pyx
missing cimport in module 'histogramnd_c': silx/math/histogramnd/chistogramnd.pyx
running build
running build_py
running build_ext
running test
WARNING:run_tests:Building silx to /Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7
missing cimport in module 'specfile': silx/io/specfile/specfile.pyx
missing cimport in module 'histogramnd_c': silx/math/histogramnd/chistogramnd.pyx
running build
running build_py
running build_ext
WARNING:run_tests:Patched sys.path, added: '/Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7'
WARNING:run_tests:Test silx 0.2.0-dev0 from /Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx
..........2016-07-04 11:25:57.695 Python[891:6816] modalSession has been exited prematurely - check for a reentrant call to endModalSession:
...F......EFF.............................................................s.............................................................................................................................................................................................................................................................................................................................................................................................................WARNING:silx.math.fit.leastsq:Supplied function does not return a 1D array of floats.
Function should be rewritten.
Trying to reshape output.
WARNING:silx.math.fit.leastsq:Supplied function does not return a 1D array of floats.
Function should be rewritten.
Trying to reshape output.
.WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
WARNING:silx.math.fit.leastsq:Recommended to set full_output to True when using constraints
./Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx/math/test/test_fit.py:51: RuntimeWarning: invalid value encountered in multiply
  return numpy.exp(x*numpy.less(abs(x), 250)) - \
WARNING:silx.math.fit.leastsq:Supplied function does not return a proper array of floats.
Function should be rewritten to return a 1D array of floats.
Trying to reshape output.
WARNING:silx.math.fit.leastsq:Supplied function unable to handle non-finite x data
Attempting to filter out those x data values.
WARNING:silx.math.fit.leastsq:Need to reshape input xdata.
...........
======================================================================
ERROR: testFailingConverters (silx.gui.plot.test.testPlotTools.TestPositionInfo)
Test PositionInfo with failing custom converters
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx/testutils.py", line 205, in wrapper
    result = func(*args, **kwargs)
  File "/Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx/testutils.py", line 161, in __exit__
    expected_count, logging.getLevelName(level), count))
RuntimeError: Expected 2 ERROR logging messages, got 0

======================================================================
FAIL: testWithAnImage (silx.gui.plot.test.testMaskToolsWidget.TestMaskToolsWidget)
Plot with an image: test MaskToolsWidget interactions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx/gui/plot/test/testMaskToolsWidget.py", line 161, in testWithAnImage
    self.assertFalse(numpy.all(numpy.equal(self.maskWidget.getMask(), 0)))
AssertionError: True is not false

======================================================================
FAIL: testAlignedProfile (silx.gui.plot.test.testPlotTools.TestProfileToolBar)
Test horizontal and vertical profile, without and with image [] (mode=Vertical Profile Mode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx/gui/plot/test/testPlotTools.py", line 171, in testAlignedProfile
    self.assertIsNot(toolButton, None)
AssertionError: unexpectedly identical: None

======================================================================
FAIL: testDiagonalProfile (silx.gui.plot.test.testPlotTools.TestProfileToolBar)
Test diagonal profile, without and with image
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/kieffer/workspace/silx/build/lib.macosx-10.11-intel-2.7/silx/gui/plot/test/testPlotTools.py", line 199, in testDiagonalProfile
    self.assertIsNot(toolButton, None)
AssertionError: unexpectedly identical: None

----------------------------------------------------------------------
Ran 495 tests in 31.323s

FAILED (failures=3, errors=1, skipped=1)
WARNING:run_tests:Test suite failed

--- a/silx/gui/plot/test/testColormapDialog.py
+++ b/silx/gui/plot/test/testColormapDialog.py
@@ -32,7 +32,7 @@ __date__ = "30/03/2016"
 import doctest
 import unittest
 
-from silx.gui.testutils import qWaitForWindowExposed
+from silx.gui.testutils import qWaitForWindowExposedAndActivate
 from silx.gui import qt
 from silx.gui.plot import ColormapDialog
 
@@ -47,7 +47,7 @@ def _tearDownQt(docTest):
     Checks that dialog widget is displayed
     """
     dialogWidget = docTest.globs['dialog']
-    qWaitForWindowExposed(dialogWidget)
+    qWaitForWindowExposedAndActivate(dialogWidget)
     dialogWidget.setAttribute(qt.Qt.WA_DeleteOnClose)
     dialogWidget.close()
     del dialogWidget

--- a/silx/gui/plot/test/testPlotTools.py
+++ b/silx/gui/plot/test/testPlotTools.py
@@ -35,7 +35,7 @@ import unittest
 
 from silx.testutils import ParametricTestCase, test_logging
 from silx.gui.testutils import (
-    qWaitForWindowExposed, TestCaseQt, getQToolButtonFromAction)
+    qWaitForWindowExposedAndActivate, TestCaseQt, getQToolButtonFromAction)
 from silx.gui import qt
 from silx.gui.plot import PlotWindow, PlotTools
 
@@ -50,7 +50,7 @@ def _tearDownDocTest(docTest):
     Checks that plot widget is displayed
     """
     plot = docTest.globs['plot']
-    qWaitForWindowExposed(plot)
+    qWaitForWindowExposedAndActivate(plot)
     plot.setAttribute(qt.Qt.WA_DeleteOnClose)
     plot.close()
     del plot

--- a/silx/gui/plot/test/testPlotWindow.py
+++ b/silx/gui/plot/test/testPlotWindow.py
@@ -33,7 +33,7 @@ import doctest
 import unittest
 
 from silx.gui.testutils import TestCaseQt, getQToolButtonFromAction
-# from silx.gui.testutils import qWaitForWindowExposed
+# from silx.gui.testutils import qWaitForWindowExposedAndActivate
 
 from silx.gui import qt
 from silx.gui.plot import PlotWindow
@@ -54,7 +54,7 @@ def _tearDownQt(docTest):
     for obj in docTest.globs.values():
         if isinstance(obj, PlotWindow):
             # Commented out as it takes too long
-            # qWaitForWindowExposed(obj)
+            # qWaitForWindowExposedAndActivate(obj)
             obj.setAttribute(qt.Qt.WA_DeleteOnClose)
             obj.close()
             del obj

--- a/silx/gui/testutils.py
+++ b/silx/gui/testutils.py
@@ -61,6 +61,26 @@ else:
     qWaitForWindowExposed = QTest.qWaitForWindowExposed
 
 
+def qWaitForWindowExposedAndActivate(window, timeout=None):
+    """Waits until the window is shown in the screen.
+
+    It also activates the window and raises it.
+
+    See QTest.qWaitForWindowExposed for details.
+    """
+    if timeout is None:
+        result = qWaitForWindowExposed(window)
+    else:
+        result = qWaitForWindowExposed(window, timeout)
+
+    if result:
+        # Makes sure window is active and on top
+        window.activateWindow()
+        window.raise_()
+
+    return result
+
+
 # Placeholder for QApplication
 _qapp = None
 
@@ -277,10 +297,7 @@ class TestCaseQt(unittest.TestCase):
 
         See QTest.qWaitForWindowExposed for details.
         """
-        if timeout is None:
-            result = qWaitForWindowExposed(window)
-        else:
-            result = qWaitForWindowExposed(window, timeout)
+        result = qWaitForWindowExposedAndActivate(window, timeout)
 
         if self.TIMEOUT_WAIT:
             QTest.qWait(self.TIMEOUT_WAIT)

